Add Pausing Cluster Queues documentation (#2127)
buildkite/docs: pages/agent/clusters.md

1. Enter a key and description.
1. Select _Create Queue_.

### Pause a queue

> Pausing a queue will prevent jobs from being dispatched to agents associated with that queue. This may affect builds targeting the paused queue.

Reviewer: ```suggestion
You can pause a queue to prevent jobs from being dispatched to agents associated with that queue. This may affect builds targeting the paused queue.
```

Just tossing around the idea of phrasing this in "You can" language like a few of the other sentences on this page, given this is an action we expect our users to take directly.

Reviewer: I also wonder if we can make this sentence clearer?

> This may affect builds targeting the paused queue.

Does this mean trigger jobs from other pipelines targeting the queue to be paused? If so, perhaps we could include that specific detail too?

Author: @timriley yeah, I wasn't sure how much detail to go into here! With trigger jobs, they don't run on agents at all so don't really care about queues being paused. If the job they're triggering relies on the paused queue, it won't run while the queue is paused. If the job they're triggering targets another queue that is not paused, then it'll go ahead and run, unless an explicit dependency has been set

Author: I've tried a best pass at explaining in more detail. Let me know what you think :)

> To pause a queue:
> 1. Navigate to your cluster’s _Queues_.

Reviewer: ```suggestion
To pause a queue:

1. Navigate to your cluster’s _Queues_.
```

I think we need this extra newline to make sure the list displays properly. Right now it's not being picked up (from the branch preview site):

<img width="778" alt="Screenshot 2023-05-22 at 1 23 16 pm" src="https://github.com/buildkite/docs/assets/3134/f334704b-a5f8-4485-b0d3-f699da9e38ce">

Author: whoops!

> 1. Click on the queue you wish to pause.
> 1. Click _Edit_.
> 1. Under _Queue Management_, click _Pause Queue_.

Reviewer: ```suggestion
1. Select the queue you wish to pause.
1. Select _Edit_.
1. Under _Queue Management_, select _Pause Queue_.
```

Just a few other "click"->"select" changes.

> 1. Enter an optional note in the dialog if needed, and confirm that you wish to pause the queue.

Reviewer 2: I think we should explain that a notice including the note will be shown on build pages affected by the pause.

Author: Good point - feel like we should mention this in the dialog itself too

> To resume the queue again, click Resume Queue.

Reviewer: ```suggestion
To resume the queue again, select _Resume Queue_.
```

We're italicising the other bits of UI text here, so this one probably needs it too?

Also, we use "select" instead of "click" as our house style for directly interacting with buttons and other UI elements.

1. Select a user or team.
1. Click _Add Maintainer_.

### Pause a queue

You can pause a queue to prevent jobs from being dispatched to agents associated with that queue.
To pause a queue:

1. Navigate to your cluster’s _Queues_.
1. Select on the queue you wish to pause.
1. Select _Edit_.
1. Under _Queue Management_, select _Pause Queue_.

> 1. Enter an optional note in the dialog if needed, and confirm that you wish to pause the queue. The note will be displayed on the queue page, and any affected builds.

Reviewer 3: I think the "if needed" might be playing another role. For me, it's hinting at the reason why you might or might not leave a note. Maybe it would be helpful to be more explicit about that? Thinking:

```suggestion
1. Enter an optional note in the dialog, and confirm that you wish to pause the queue. 
   
    You can use the note to explain why you're pausing the queue. The note will display on the queue page and any affected builds.
```

Note, the spacing should display the clarifying sentence indented with the list. I think I've got the spacing right, but worth double checking.

> Jobs which have _already_ been dispatched to agents in the queue prior to pausing will continue to run. New jobs which target the paused queue will 'wait' until the queue is resumed.

Reviewer 3: This could be further tightened:

```suggestion
Jobs _already_ dispatched to agents in the queue before pausing will continue to run. New jobs that target the paused queue will wait until the queue is resumed.
```

But, being more verbose could help with clarity. You can decide which is the better way to go here. :)

> Trigger steps do not rely on agents, so they will run unless they have a dependency that has been halted by the paused queue. The behaviour of the jobs they trigger depends on their configuration. If a triggered job targets the paused queue, it will 'wait' until the queue is resumed. If a triggered job does not target the paused queue, it will run as usual.

Reviewer: Also a good clarifying note!

Just took the quotes off "wait" here too:

```suggestion
Trigger steps do not rely on agents, so they will run unless they have a dependency that has been halted by the paused queue. The behaviour of the jobs they trigger depends on their configuration. If a triggered job targets the paused queue, it will wait until the queue is resumed. If a triggered job does not target the paused queue, it will run as usual.
```

My only other thought here is whether "halted" is the right term. We're not using it anywhere else in this section. I wonder if we could rephrase that first sentence to avoid it...

> Trigger steps do not rely on agents, so they will run unless they have dependencies that are also waiting with the paused queue.

Maybe?

Reviewer 3: +1 to avoiding adding a new term here if we can avoid it. Another option for tightening:

> Trigger steps do not rely on agents, so they will run unless they have dependencies waiting on the paused queue.

> Trigger steps do not rely on agents, so they will run unless they have a dependency that has been halted by the paused queue. The behaviour of the jobs they trigger depends on their configuration. If a triggered job targets the paused queue, it will 'wait' until the queue is resumed. If a triggered job does not target the paused queue, it will run as usual.

Reviewer 3: Use US English:

```suggestion
Trigger steps do not rely on agents, so they will run unless they have a dependency that has been halted by the paused queue. The behavior of the jobs they trigger depends on their configuration. If a triggered job targets the paused queue, it will 'wait' until the queue is resumed. If a triggered job does not target the paused queue, it will run as usual.
```

> To resume the queue again, select _Resume Queue_. Once resumed, job dispatch to the queue will operate as usual, and any 'waiting' jobs affected by the pause will be picked up.

Reviewer: ```suggestion
Select _Resume Queue_ to resume a paused queue. Job dispatch to the queue will operate as usual and any previously waiting jobs will be run.
```

Reviewer 3: I'll add a further suggestion to the second sentence:

```suggestion
Select _Resume Queue_ to resume a paused queue. Jobs will resume dispatching to the queue as usual, including any jobs waiting to run.
```
